### ISSUE TITLE
Rule proposal: No unnecessary check for falsy values

### ISSUE THREAD

**Issue by mohsen1** [typescript-eslint/typescript-eslint#356]
Code like this is probably result of wrong type annotation:

```ts
function foo(bar: string[]) {
 if (bar) {
   return bar.join(' -- ')
  }

  return '';
}
```

in my case type annotation was wrong and `bar` had to be of type `string[] | null`. 

We should detect those checks that are always truthy according to types and warn against them.

**Comment by threehams** [comment]
Is this accepting PRs? I would be very happy to work on this, as an unnecessary falsy check can mean someone has done one-off protection against incorrect types rather than fixing the underlying problem.

**Comment by bradzacher** [comment]
This is handled by the `no-unnecessary-condition` rule.
Sorry, the issue didn't get linked to the PR.
